javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/matcher/AgentElementMatchers.java

```diff
@@ -36,8 +35,7 @@
 
   public static ElementMatcher.Junction<TypeDescription> safeHasSuperType(
       ElementMatcher<TypeDescription> matcher) {
-    return not(isInterface())
-        .and(new SafeHasSuperTypeMatcher(new SafeErasureMatcher<>(matcher), false));
+    return new SafeHasSuperTypeMatcher(new SafeErasureMatcher<>(matcher), false);
```

Comment:
Without this change, interface default methods are not instrumented. I looked through the callers and changed some of them to `extendsClass()`, and the remaining seem more correct taking interface default methods in to account.